modules/currency-l0/src/main/scala/org/tessellation/currency/l0/CurrencyL0App.scala

```diff
@@ -45,7 +45,7 @@ abstract class CurrencyL0App(
   val kryoRegistrar: Map[Class[_], KryoRegistrationId[KryoRegistrationIdRange]] =
     sdkKryoRegistrar
 
-  def dataApplication: Option[BaseDataApplicationL0Service[IO]]
+  def dataApplication: IO[Option[BaseDataApplicationL0Service[IO]]] = none.pure[IO]
```

Comment:
The data application should be wrapped into IO and the whole implementation should be optional, so it should be: Option[IO

Reply:
Why is that the desired interface?

Reply:
Because providing the data application is optional for metagraph developer. One can just create a metagraph with just regular currency transactions without any custom data. So the `Option` should be the base effect for that def.

Reply:
changed